libevmasm/ConstantOptimiser.cpp

```diff
@@ -253,7 +253,47 @@ AssemblyItems ComputeMethod::findRepresentation(u256 const& _value)
 	if (_value < 0x10000)
 		// Very small value, not worth computing
 		return AssemblyItems{_value};
-	else if (numberEncodingSize(~_value) < numberEncodingSize(_value))
+
+	// check for masks first
+	unsigned lowZeros = 0;
+	unsigned highOnes = 0;
+	for (; ((_value >> lowZeros) & 1) == 0 && lowZeros < 256; lowZeros++) {}
+	for (; ((_value >> (lowZeros + highOnes)) & 1) == 1 && highOnes < 256; highOnes++) {}
+	if (m_params.evmVersion.hasBitwiseShifting() && highOnes > 32 &&
+		((_value >> (lowZeros + highOnes)) == 0) &&
+		((lowZeros + highOnes < 256) || lowZeros > 16))
```

Comment:
This is how we wrap long conditions:

```suggestion
	if (
		m_params.evmVersion.hasBitwiseShifting() &&
		highOnes > 32 &&
		(_value >> (lowZeros + highOnes)) == 0 &&
		(lowZeros + highOnes < 256 || lowZeros > 16)
	)
```

Same for the other `if` below.

Reply:
also added comments on each condition